### ISSUE TITLE
Add conditions to DataVolume

### ISSUE THREAD

**Issue by rmohr** [kubevirt/containerized-data-importer#893]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement

**What happened**:

As a follow-up of #892: Can you expose some conditions which indicate why an image upload failed? Right now we only create a generic warning that the DataVolume failed but we can't send why it actually failed. If you set the condition we can send specific events and set the condition also on the VMI.

**What you expected to happen**:

Add conditions so that people can understand why the import has issues.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`):
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Install tools:
- Others:

**Comment by moadqassem** [comment]
Sending those events/conditions to the VMI wouldn't be sufficient, because the VMI is actually dependent on the DataVolume, in other words, if a DataVolume fails no VMI will be created, thus those events should be sent to the VM instead.

**Comment by awels** [comment]
As CDI, we have no concept of VM/VMIs at all. So we are not going to send events there. We should however be creating events on the data volume so that whomever is interested in them can read them. We are currently sending those events to the underlying PVC of a data volume, but haven't gotten to also creating them for the DV.

**Comment by rmohr** [comment]
> As CDI, we have no concept of VM/VMIs at all.

I think it would still make sense to have conditions on the DataVolume which we can read. We can then do more specific events on the VM/VMI, if these conditions contain useful information. Right now we just send a generic warning that "something" is wrong with the DataVolume.

**Comment by alvaroaleman** [comment]
> We are currently sending those events to the underlying PVC of a data volume, but haven't gotten to also creating them for the DV.

Yeah that would be much more intuitive, I would only only expect events from the actual PV provisioning on the PV. And the CDI does IMHO not count as "PV provisioning", because it has its own custom resource and PVs are just a subordinate type from its POV

**Comment by awels** [comment]
/remove-lifecycle-stale
We haven't gotten around to this, but it is important.

**Comment by awels** [comment]
https://github.com/kubevirt/containerized-data-importer/pull/1194 should resolve the DV side of things. We added conditions with one of them reporting the exit message of the cdi import/clone/upload pod.
